techniques/applications/packageManagement/1.1/packageManagement.st

```diff
@@ -0,0 +1,139 @@
+
+      # Build string vars used for reporting
+
+      # State
+      "state_description[${index_pkg}]" string     => "Presence",
+                                        ifvarclass => "state_present_${index_pkg}";
+      "state_description[${index_pkg}]" string     => "Absence",
+                                        ifvarclass => "!state_present_${index_pkg}";
+
+      # Architecture
+      "architecture_description[${index_pkg}]" string     => " for ${architecture[${index_pkg}]} architecture ",
+                                               ifvarclass => "architecture_specified_${index_pkg}";
+      "architecture_description[${index_pkg}]" string     => " ",
+                                               ifvarclass => "!architecture_specified_${index_pkg}";
+
+      # Version
+      "version_description[${index_pkg}]" string     => "in latest available version",
+                                          ifvarclass => "version_latest_${index_pkg}";
+      "version_description[${index_pkg}]" string     => "in any version",
+                                          ifvarclass => "!version_latest_${index_pkg}.!version_specified_${index_pkg}";
+      "version_description[${index_pkg}]" string     => "in version ${version[${index_pkg}]}",
+                                          ifvarclass => "version_specified_${index_pkg}";
+
+      # Allow untrusted
+      "manager_final_options[${index_pkg}]" string => "-o Apt::Get::AllowUnauthenticated=true ${manager_options[${index_pkg}]}",
+                                        ifvarclass => "manager_apt_${index_pkg}.allow_unstrusted_${index_pkg}";
+      "manager_final_options[${index_pkg}]" string => "${manager_options[${index_pkg}]}",
+                                        ifvarclass => "!manager_apt_${index_pkg}|!allow_unstrusted_${index_pkg}";
+
+      # Message for old agents
+      "incompatible_error" string => "This technique is not compatible with Rudder 3.1 or older. Skipping.";
+
+      # Name of the ncf bundle. We use a variable to avoid breaking the syntax if the bundle does not exist.
+      "bundle_name" string => "package_state_options";
+   
+  classes:
+      # This class is different from the one in ncf (where a version number and latest are considered "specified")
+      "version_specified_${index_pkg}" expression => strcmp("${version[${index_pkg}]}", "specific");
+      "architecture_specified_${index_pkg}" expression => strcmp("${architecture[${index_pkg}]}", "specific");
+      "posthook_specified_${index_pkg}" not => strcmp("${posthook[${index_pkg}]}", "");
+      "state_present_${index_pkg}" expression => strcmp("${state[${index_pkg}]}", "present");
+      "version_latest_${index_pkg}" expression => strcmp("${version[${index_pkg}]}", "latest");
+      "allow_unstrusted_${index_pkg}" expression => strcmp("${manager_allow_untrusted[${index_pkg}]}", "true");
+      "manager_apt_${index_pkg}" expression => strcmp("${manager[${index_pkg}]}", "apt");
```

Comment:
We could duplicate the logic for default package manager selection here, but I don't like it either.